added the ability to reply to comments and @ mention other users

rahulchhabria opened the pull request:
work for: https://github.com/dcramer/peated/issues/194

rahulchhabria committed:
added the ability to reply to comments and @ mention other users

dcramer commented on apps/server/migrations/0200_add_comment_replies.sql:
```diff
@@ -0,0 +1,2 @@
+-- Add parent_id column to comments table
+ALTER TABLE "comments" ADD COLUMN IF NOT EXISTS "parent_id" bigint REFERENCES "comments"("id"); 
```
you can ignore this coment - while yes its probably accurate, we may not want comments to be fully deletable. youd probably want a delete comment to just show it was deleted, and hide a bunch of info about it

rahulchhabria replied:
<img width="1050" alt="Screenshot 2025-03-14 at 11 57 45 PM" src="https://github.com/user-attachments/assets/05ea80b7-3cb4-4941-ab17-b0f5aa0954f0" />

dcramer commented on apps/server/migrations/0202_add_mentions_to_comments.sql:
```diff
@@ -0,0 +1,2 @@
+-- Add mentions field to comments table with a length constraint
+ALTER TABLE "comments" ADD COLUMN IF NOT EXISTS "mentions" varchar(500); 
```
you'd probably want a varchar[] type field here which is hard mode

dcramer replied:
actually the complexity here is you can change your username so really what you'd want is something more annoying like mentioned user ids but you kind of need a mention username to id mapping

could just replace the mentions in the comments with some special syntax which is the ID, then replace them on output as well? not totally sure

dcramer commented on apps/server/src/db/schema/comments.ts:
```diff
@@ -6,24 +6,35 @@ import {
+      { onDelete: "cascade" },
+    ),
+    // Add mentions field to store mentioned usernames with a length constraint
+    mentions: varchar("mentions", { length: 500 }),
```
look at tags schema for an example of how to use a varchar array (not sure if my default is right there)

dcramer commented on apps/server/src/schemas/comments.ts:
```diff
@@ -2,14 +2,42 @@ import { z } from "zod";
+    ),
   createdAt: z.string().datetime(),
   createdBy: UserSchema,
+  replyToId: z.number().nullable().optional(),
```
id prob make this replyTo (which is just the lazy pattern i use elsewhere in schemas)

rahulchhabria committed:
better handling of deleted comments

now a deleted comment will remain in the ui but the comment will be replaced with "this comment was deleted"
